schematic/schemas/data_model_graph.py

```diff
@@ -226,6 +229,64 @@ def get_adjacent_nodes_by_relationship(
 
         return list(nodes)
 
+    def get_component_node_validation_rules(
+        self, manifest_component: str, node_display_name: str
+    ) -> list[str]:
+        """Get valdation rules for a given node and component.
+        Args:
+            manifest_component, str: manifest component display name that the node belongs to.
+            node_display_name, str: node display name for the node being queried.
+        Returns:
+            validation_rules, list[str]: validation rules list for a given node and component.
+        """
+        # get any additional validation rules associated with this node (e.g. can this node be mapped to a list of other nodes)
+        node_validation_rules = self.get_node_validation_rules(
+            node_display_name=node_display_name
+        )
+
+        # Parse the validation rules per component if applicable
+        if node_validation_rules and isinstance(node_validation_rules, dict):
+            node_validation_rules = extract_component_validation_rules(
```

Comment:
The return type from `get_node_validation_rules` is a list, your'e checking if `node_validation_rules` is a dict

Reply:
Ah this is bc in between my original writing of this function, and the implementation of component based validation rules, we introduced validation rule being recorded as dictionaries as well as lists... so ill need to update that typing.